### PR TITLE
add kitex binding

### PR DESCRIPTION
**Pull request by songoow**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

add kitex binding 
## Issue reference
https://github.com/dapr/docs/issues/3266

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+title: "Kitex"
+linkTitle: "Kitex"
+description: "Detailed documentation on the Kitex binding component"
+aliases:
+- "/operations/components/setup-bindings/supported-bindings/kitex/"
+---
+
+## Overview
+
+The binding for kitex mainly utilizes the generic-call feature in kitex. More information can be found in [kitex generic-call](https://www.cloudwego.io/docs/kitex/tutorials/advanced-feature/generic-call/).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
The binding for Kitex mainly utilizes the generic-call feature in Kitex. Learn more from the official documentation around [Kitex generic-call](https://www.cloudwego.io/docs/kitex/tutorials/advanced-feature/generic-call/).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+Currently, kitex only supports Thrift generic calls, and the implementation integrated into [components-contrib](https://github.com/dapr/components-contrib/tree/master/bindings/kitex) adopts binary generic calls.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
Currently, Kitex only supports Thrift generic calls. The implementation integrated into [components-contrib](https://github.com/dapr/components-contrib/tree/master/bindings/kitex) adopts binary generic calls.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+
+## Component format
+
+To setup an Kitex binding create a component of type `bindings.kitex`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
To setup an Kitex binding, create a component of type `bindings.kitex`. See the [How-to: Use output bindings to interface with external resources]({{< ref "howto-bindings.md#1-create-a-binding" >}}) guide on creating and applying a binding configuration.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  name: bindings.kitex
+spec:
+  type: bindings.kitex
+  version: v1
+  metadata: 
+```
+
+## Spec metadata fields
+
+The `InvokeRequest.Metadata` for `bindings.kitex` requires the client to fill in four required items: hostPorts, destService, methodName and version when making a call.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
The `InvokeRequest.Metadata` for `bindings.kitex` requires the client to fill in four required items when making a call: 
- `hostPorts`
- `destService`
- `methodName`
- `version` 
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+| Field       | Required | Binding support | Details                                                                                                 | Example            |
+|-------------|:--------:|--------|---------------------------------------------------------------------------------------------------------|--------------------|
+| hostPorts   |    Y     | Output | IP address and port information of the kitex server (thrift)                                        | `"127.0.0.1:8888"` |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
| hostPorts   |    Y     | Output | IP address and port information of the Kitex server (Thrift)                                        | `"127.0.0.1:8888"` |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+| destService |    Y     | Output | Service name of the kitex server (thrift)            | `"echo"`           |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
| destService |    Y     | Output | Service name of the Kitex server (Thrift)            | `"echo"`           |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+| methodName  |    Y     | Output | Method name under a specific service name of the kitex server (thrift) | `"echo"`           |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
| methodName  |    Y     | Output | Method name under a specific service name of the Kitex server (Thrift) | `"echo"`           |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+| version     |    Y     | Output | kitex version                                                                                           | `"0.5.0"`          |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
| version     |    Y     | Output | Kitex version                                                                                           | `"0.5.0"`          |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+## Binding support
+
+This component supports **output binding** with the following operations:
+
+- `get`
+
+## Example 
+
+When using kitex binding, the client needs to pass in the correct Thrift-encoded binary, and the server needs to be a Thrift Server. The [kitex_output_test](https://github.com/dapr/components-contrib/blob/master/bindings/kitex/kitex_output_test.go) can be used as a reference.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
When using Kitex binding:
- The client needs to pass in the correct Thrift-encoded binary
- The server needs to be a Thrift Server. 

The [kitex_output_test](https://github.com/dapr/components-contrib/blob/master/bindings/kitex/kitex_output_test.go) can be used as a reference.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+For example, the variable `reqData` needs to be encoded by the Thrift protocol before sending, and the returned data needs to be decoded by the Thrift protocol.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kitex.md]:
```suggestion
For example, the variable `reqData` needs to be _encoded_ by the Thrift protocol before sending, and the returned data needs to be decoded by the Thrift protocol.
```
```suggestion
For example, the variable `reqData` needs to be _encoded_ by the Thrift protocol before sending, and the returned data needs to be then _decoded_ by the Thrift protocol.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review

**Comment by songoow** [comment]
@hhunter-ms Thanks, these modifications have made the document clearer.

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**